// @elastic/elasticsearch-serverless/lib/api/index.js
"use strict";
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u;
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
/* eslint-disable import/export */
/* eslint-disable @typescript-eslint/no-misused-new */
/* eslint-disable @typescript-eslint/no-extraneous-class */
/* eslint-disable @typescript-eslint/no-unused-vars */
// This file was automatically generated by elastic/elastic-client-generator-js
// DO NOT MODIFY IT BY HAND. Instead, modify the source open api file,
// and elastic/elastic-client-generator-js to regenerate this file again.
const async_search_1 = tslib_1.__importDefault(require("./api/async_search"));
const bulk_1 = tslib_1.__importDefault(require("./api/bulk"));
const cat_1 = tslib_1.__importDefault(require("./api/cat"));
const clear_scroll_1 = tslib_1.__importDefault(require("./api/clear_scroll"));
const close_point_in_time_1 = tslib_1.__importDefault(require("./api/close_point_in_time"));
const cluster_1 = tslib_1.__importDefault(require("./api/cluster"));
const count_1 = tslib_1.__importDefault(require("./api/count"));
const create_1 = tslib_1.__importDefault(require("./api/create"));
const delete_1 = tslib_1.__importDefault(require("./api/delete"));
const delete_by_query_1 = tslib_1.__importDefault(require("./api/delete_by_query"));
const delete_script_1 = tslib_1.__importDefault(require("./api/delete_script"));
const enrich_1 = tslib_1.__importDefault(require("./api/enrich"));
const eql_1 = tslib_1.__importDefault(require("./api/eql"));
const exists_1 = tslib_1.__importDefault(require("./api/exists"));
const exists_source_1 = tslib_1.__importDefault(require("./api/exists_source"));
const explain_1 = tslib_1.__importDefault(require("./api/explain"));
const field_caps_1 = tslib_1.__importDefault(require("./api/field_caps"));
const get_1 = tslib_1.__importDefault(require("./api/get"));
const get_script_1 = tslib_1.__importDefault(require("./api/get_script"));
const get_source_1 = tslib_1.__importDefault(require("./api/get_source"));
const graph_1 = tslib_1.__importDefault(require("./api/graph"));
const index_1 = tslib_1.__importDefault(require("./api/index"));
const indices_1 = tslib_1.__importDefault(require("./api/indices"));
const inference_1 = tslib_1.__importDefault(require("./api/inference"));
const info_1 = tslib_1.__importDefault(require("./api/info"));
const ingest_1 = tslib_1.__importDefault(require("./api/ingest"));
const license_1 = tslib_1.__importDefault(require("./api/license"));
const logstash_1 = tslib_1.__importDefault(require("./api/logstash"));
const mget_1 = tslib_1.__importDefault(require("./api/mget"));
const ml_1 = tslib_1.__importDefault(require("./api/ml"));
const msearch_1 = tslib_1.__importDefault(require("./api/msearch"));
const msearch_template_1 = tslib_1.__importDefault(require("./api/msearch_template"));
const mtermvectors_1 = tslib_1.__importDefault(require("./api/mtermvectors"));
const open_point_in_time_1 = tslib_1.__importDefault(require("./api/open_point_in_time"));
const ping_1 = tslib_1.__importDefault(require("./api/ping"));
const put_script_1 = tslib_1.__importDefault(require("./api/put_script"));
const query_ruleset_1 = tslib_1.__importDefault(require("./api/query_ruleset"));
const rank_eval_1 = tslib_1.__importDefault(require("./api/rank_eval"));
const reindex_1 = tslib_1.__importDefault(require("./api/reindex"));
const render_search_template_1 = tslib_1.__importDefault(require("./api/render_search_template"));
const scripts_painless_execute_1 = tslib_1.__importDefault(require("./api/scripts_painless_execute"));
const scroll_1 = tslib_1.__importDefault(require("./api/scroll"));
const search_1 = tslib_1.__importDefault(require("./api/search"));
const search_application_1 = tslib_1.__importDefault(require("./api/search_application"));
const search_mvt_1 = tslib_1.__importDefault(require("./api/search_mvt"));
const search_template_1 = tslib_1.__importDefault(require("./api/search_template"));
const security_1 = tslib_1.__importDefault(require("./api/security"));
const sql_1 = tslib_1.__importDefault(require("./api/sql"));
const synonyms_1 = tslib_1.__importDefault(require("./api/synonyms"));
const tasks_1 = tslib_1.__importDefault(require("./api/tasks"));
const terms_enum_1 = tslib_1.__importDefault(require("./api/terms_enum"));
const termvectors_1 = tslib_1.__importDefault(require("./api/termvectors"));
const transform_1 = tslib_1.__importDefault(require("./api/transform"));
const update_1 = tslib_1.__importDefault(require("./api/update"));
const update_by_query_1 = tslib_1.__importDefault(require("./api/update_by_query"));
const kAsyncSearch = Symbol('AsyncSearch');
const kCat = Symbol('Cat');
const kCluster = Symbol('Cluster');
const kEnrich = Symbol('Enrich');
const kEql = Symbol('Eql');
const kGraph = Symbol('Graph');
const kIndices = Symbol('Indices');
const kInference = Symbol('Inference');
const kIngest = Symbol('Ingest');
const kLicense = Symbol('License');
const kLogstash = Symbol('Logstash');
const kMl = Symbol('Ml');
const kQueryRuleset = Symbol('QueryRuleset');
const kSearchApplication = Symbol('SearchApplication');
const kSecurity = Symbol('Security');
const kSql = Symbol('Sql');
const kSynonyms = Symbol('Synonyms');
const kTasks = Symbol('Tasks');
const kTransform = Symbol('Transform');
class API {
    constructor() {
        Object.defineProperty(this, _a, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _b, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _c, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _d, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _e, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _f, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _g, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _h, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _j, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _k, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _l, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _m, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _o, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _p, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _q, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _r, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _s, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _t, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, _u, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this[kAsyncSearch] = null;
        this[kCat] = null;
        this[kCluster] = null;
        this[kEnrich] = null;
        this[kEql] = null;
        this[kGraph] = null;
        this[kIndices] = null;
        this[kInference] = null;
        this[kIngest] = null;
        this[kLicense] = null;
        this[kLogstash] = null;
        this[kMl] = null;
        this[kQueryRuleset] = null;
        this[kSearchApplication] = null;
        this[kSecurity] = null;
        this[kSql] = null;
        this[kSynonyms] = null;
        this[kTasks] = null;
        this[kTransform] = null;
    }
}
_a = kAsyncSearch, _b = kCat, _c = kCluster, _d = kEnrich, _e = kEql, _f = kGraph, _g = kIndices, _h = kInference, _j = kIngest, _k = kLicense, _l = kLogstash, _m = kMl, _o = kQueryRuleset, _p = kSearchApplication, _q = kSecurity, _r = kSql, _s = kSynonyms, _t = kTasks, _u = kTransform;
exports.default = API;
API.prototype.bulk = bulk_1.default;
API.prototype.clearScroll = clear_scroll_1.default;
API.prototype.closePointInTime = close_point_in_time_1.default;
API.prototype.count = count_1.default;
API.prototype.create = create_1.default;
API.prototype.delete = delete_1.default;
API.prototype.deleteByQuery = delete_by_query_1.default;
API.prototype.deleteScript = delete_script_1.default;
API.prototype.exists = exists_1.default;
API.prototype.existsSource = exists_source_1.default;
API.prototype.explain = explain_1.default;
API.prototype.fieldCaps = field_caps_1.default;
API.prototype.get = get_1.default;
API.prototype.getScript = get_script_1.default;
API.prototype.getSource = get_source_1.default;
API.prototype.index = index_1.default;
API.prototype.info = info_1.default;
API.prototype.mget = mget_1.default;
API.prototype.msearch = msearch_1.default;
API.prototype.msearchTemplate = msearch_template_1.default;
API.prototype.mtermvectors = mtermvectors_1.default;
API.prototype.openPointInTime = open_point_in_time_1.default;
API.prototype.ping = ping_1.default;
API.prototype.putScript = put_script_1.default;
API.prototype.rankEval = rank_eval_1.default;
API.prototype.reindex = reindex_1.default;
API.prototype.renderSearchTemplate = render_search_template_1.default;
API.prototype.scriptsPainlessExecute = scripts_painless_execute_1.default;
API.prototype.scroll = scroll_1.default;
API.prototype.search = search_1.default;
API.prototype.searchMvt = search_mvt_1.default;
API.prototype.searchTemplate = search_template_1.default;
API.prototype.termsEnum = terms_enum_1.default;
API.prototype.termvectors = termvectors_1.default;
API.prototype.update = update_1.default;
API.prototype.updateByQuery = update_by_query_1.default;
Object.defineProperties(API.prototype, {
    asyncSearch: {
        get() { return this[kAsyncSearch] === null ? (this[kAsyncSearch] = new async_search_1.default(this.transport)) : this[kAsyncSearch]; }
    },
    cat: {
        get() { return this[kCat] === null ? (this[kCat] = new cat_1.default(this.transport)) : this[kCat]; }
    },
    cluster: {
        get() { return this[kCluster] === null ? (this[kCluster] = new cluster_1.default(this.transport)) : this[kCluster]; }
    },
    enrich: {
        get() { return this[kEnrich] === null ? (this[kEnrich] = new enrich_1.default(this.transport)) : this[kEnrich]; }
    },
    eql: {
        get() { return this[kEql] === null ? (this[kEql] = new eql_1.default(this.transport)) : this[kEql]; }
    },
    graph: {
        get() { return this[kGraph] === null ? (this[kGraph] = new graph_1.default(this.transport)) : this[kGraph]; }
    },
    indices: {
        get() { return this[kIndices] === null ? (this[kIndices] = new indices_1.default(this.transport)) : this[kIndices]; }
    },
    inference: {
        get() { return this[kInference] === null ? (this[kInference] = new inference_1.default(this.transport)) : this[kInference]; }
    },
    ingest: {
        get() { return this[kIngest] === null ? (this[kIngest] = new ingest_1.default(this.transport)) : this[kIngest]; }
    },
    license: {
        get() { return this[kLicense] === null ? (this[kLicense] = new license_1.default(this.transport)) : this[kLicense]; }
    },
    logstash: {
        get() { return this[kLogstash] === null ? (this[kLogstash] = new logstash_1.default(this.transport)) : this[kLogstash]; }
    },
    ml: {
        get() { return this[kMl] === null ? (this[kMl] = new ml_1.default(this.transport)) : this[kMl]; }
    },
    queryRuleset: {
        get() { return this[kQueryRuleset] === null ? (this[kQueryRuleset] = new query_ruleset_1.default(this.transport)) : this[kQueryRuleset]; }
    },
    searchApplication: {
        get() { return this[kSearchApplication] === null ? (this[kSearchApplication] = new search_application_1.default(this.transport)) : this[kSearchApplication]; }
    },
    security: {
        get() { return this[kSecurity] === null ? (this[kSecurity] = new security_1.default(this.transport)) : this[kSecurity]; }
    },
    sql: {
        get() { return this[kSql] === null ? (this[kSql] = new sql_1.default(this.transport)) : this[kSql]; }
    },
    synonyms: {
        get() { return this[kSynonyms] === null ? (this[kSynonyms] = new synonyms_1.default(this.transport)) : this[kSynonyms]; }
    },
    tasks: {
        get() { return this[kTasks] === null ? (this[kTasks] = new tasks_1.default(this.transport)) : this[kTasks]; }
    },
    transform: {
        get() { return this[kTransform] === null ? (this[kTransform] = new transform_1.default(this.transport)) : this[kTransform]; }
    }
});
